patches/net.minecraft.client.settings.GameSettings.java.patch

```diff
@@ -1,6 +1,64 @@
++
++import java.io.*;
+ import java.lang.reflect.ParameterizedType;
+ import java.lang.reflect.Type;
++import java.nio.file.Files;
++import java.nio.file.StandardCopyOption;
+ import java.util.List;
+ import java.util.Map;
+ import java.util.Set;
+@@ -32,6 +31,7 @@
+ import org.apache.commons.lang3.ArrayUtils;
+ import org.apache.logging.log4j.LogManager;
+ import org.apache.logging.log4j.Logger;
++import org.lwjgl.Sys;
+ import org.lwjgl.input.Keyboard;
+ import org.lwjgl.input.Mouse;
+ import org.lwjgl.opengl.Display;
+@@ -55,7 +55,7 @@
+    };
+    /** GUI scale values */
+    private static final String[] GUISCALES = new String[]{"options.guiScale.auto", "options.guiScale.small", "options.guiScale.normal", "options.guiScale.large"};
+-   private static final String[] PARTICLES = new String[]{"options.particles.all", "options.particles.decreased", "options.particles.minimal"};
++   private static final String[] PARTICLES = new String[]{"options.particles.all", "options.particles.decreased", "options.particles.minimal", "options.particles.disabled"};
+    private static final String[] AMBIENT_OCCLUSIONS = new String[]{"options.ao.off", "options.ao.min", "options.ao.max"};
+    private static final String[] STREAM_COMPRESSIONS = new String[]{"options.stream.compression.low", "options.stream.compression.medium", "options.stream.compression.high"};
+    private static final String[] STREAM_CHAT_MODES = new String[]{"options.stream.chat.enabled.streaming", "options.stream.chat.enabled.always", "options.stream.chat.enabled.never"};
+@@ -169,7 +169,7 @@
+    public String language;
+    public boolean forceUnicodeFont;
+ 
+-   public GameSettings(Minecraft mcIn, File optionsFileIn) {
++   public GameSettings(Minecraft mcIn, File optionsFileIn) throws IOException {
+       this.keyBindings = (KeyBinding[])ArrayUtils.addAll(new KeyBinding[]{this.keyBindAttack, this.keyBindUseItem, this.keyBindForward, this.keyBindLeft, this.keyBindBack, this.keyBindRight, this.keyBindJump, this.keyBindSneak, this.keyBindSprint, this.keyBindDrop, this.keyBindInventory, this.keyBindChat, this.keyBindPlayerList, this.keyBindPickBlock, this.keyBindCommand, this.keyBindScreenshot, this.keyBindTogglePerspective, this.keyBindSmoothCamera, this.keyBindStreamStartStop, this.keyBindStreamPauseUnpause, this.keyBindStreamCommercials, this.keyBindStreamToggleMic, this.keyBindFullscreen, this.keyBindSpectatorOutlines}, this.keyBindsHotbar);
+       this.difficulty = EnumDifficulty.NORMAL;
+       this.lastServer = "";
+@@ -177,7 +177,12 @@
+       this.language = "en_US";
+       this.forceUnicodeFont = false;
+       this.mc = mcIn;
+-      this.optionsFile = new File(optionsFileIn, "options.txt");
++      File def = new File(optionsFileIn, "options.txt");
++      this.optionsFile = new File(optionsFileIn, "options-hyperium.txt");
++      if(!optionsFile.exists()) {
++         Files.copy(def.toPath(), optionsFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
```

Comment:
does it actually read from this file on startup rather than the options.txt file?

Reply:
yes it does, that's how mc works (unless this is put in the wrong place in which case it doesnt)

Reply:
any change made to options.txt gets copied over to the options-hyperium.txt file which eliminates the purpose of it. it's meant to only copy over once, then read from the options-hyperium.txt file from then on.

Reply:
actually it only copies over if options-hyperium.txt gets deleted

Reply:
incorrect, just tested, changed options.txt to max framerate at 5, ran the game again, options-hyperium.txt changed to also have 5fps

Reply:
bruh how that clearly says `if (!optionsFile.exists())` java's high or something

Reply:
you can try it on your own if you'd like, but it doesnt follow what's meant to be followed

Reply:
does not happen

Reply:
bruh